app/controllers/concerns/hyrax/embargoes_controller_behavior.rb

```diff
@@ -24,23 +24,29 @@ def destroy
     end
 
     # Updates a batch of embargos
+    # rubocop:disable Metrics/AbcSize
+    # rubocop:disable Metrics/MethodLength
     def update
       filter_docs_with_edit_access!
       copy_visibility = []
       copy_visibility = params[:embargoes].values.map { |h| h[:copy_visibility] } if params[:embargoes]
-      af_objects = Hyrax.custom_queries.find_many_by_alternate_ids(alternate_ids: batch, use_valkyrie: false)
-      af_objects.each do |curation_concern|
-        Hyrax::Actors::EmbargoActor.new(curation_concern).destroy
+      resources = Hyrax.custom_queries.find_many_by_alternate_ids(alternate_ids: batch, use_valkyrie: Hyrax.config.use_valkyrie?)
+      resources.each do |resource|
+        EmbargoManager.new(resource: resource).release! if Hyrax.config.use_valkyrie?
+        resource = Wings::Valkyrie::ResourceFactory.new(adapter: Wings::Valkyrie::MetadataAdapter).from_resource(resource: resource) if Hyrax.config.use_valkyrie?
```

Comment:
i'm not sure i'm following this. what's gonig on here?

Reply:
This transforms the object back into an activefedora object. Then we can use that to expire the embargo on the af side of things. 

When an embargo is expired and you try to persist the valkyrie object, it rejects saving the object because the `Embargo date needs to be set in the future`. So since we aren't really trying to save, just expire the embargo, we can just use the old code and operate on a valkyrie object if use_valkyrie is set to true, and then expire the embargo for the af object as per usual.

Reply:
>  Then we can use that to expire the embargo on the af side of things.

ah, i think this is where i got lost. i don't think we want to do two transformations (which are time-wise costly) to do this work.

>  it rejects saving the object because the Embargo date needs to be set in the future

let's focus in on this as undesirable behavior. can you work up a failing test demonstrating this case?